api/src/opentrons/motion_planning/deck_conflict.py

```diff
@@ -210,10 +217,12 @@ def check(
             )
 
 
-def _create_ot2_restrictions(
-    item: DeckItem, location: DeckSlotName
+def _create_ot2_restrictions(  # noqa: C901
+    item: DeckItem, location: Union[DeckSlotName, StagingSlotName]
 ) -> List[_DeckRestriction]:
     restrictions: List[_DeckRestriction] = []
+    if isinstance(location, StagingSlotName):
+        raise DeckConflictError("OT-2 does not support staging slots.")
```

Comment:
Let's include the problematic slot in the message.

```suggestion
    if isinstance(location, StagingSlotName):
        raise DeckConflictError(f"OT-2 does not support staging slots ({location.id}).")
```